### ISSUE TITLE
kafka offset skips messages that have not finished getting handled when using concurrency

### ISSUE THREAD

**Issue by Natanor** [robinhood/faust#355]
## Checklist

- [x] I have included information about relevant versions
- [ ] I have verified that the issue persists when using the `master` branch of Faust.

## Steps to reproduce
When running an agent with concurrency > 1, where the time it takes for some messages to get handled is vastly different than others.

the kafka topic in this example has the numbers 1 to 300 in order.
this is the consumer:
```python
sync def some_calculation(i):
    print(i)
    if i == 1:
        return
    if i == 50:
        return
    if i == 10:
        await asyncio.sleep(5)
        return
    await asyncio.sleep(100000)


@app.agent(input_topic, concurrency=100)
async def join(nums):
    async for num in nums:
        await some_calculation(num)
```

## Expected behavior

I expect the topic offset to be raised once because the first message gets finished, but then the offset to stay at 0 for 100000 seconds because the second message doesn't finish handling

## Actual behavior

at first, the offset is indeed set to 0, but after 5 seconds, it is set to 9, and after 3-ish more seconds,
it is set to 49.

## Suggested Fix

I think the fix is to assure the list of acknowledged messages in the `_new_offset` function in faust/transport/consumer.py doesn't have a gap between its start and the current offset.

# Versions

* Python version 3,6,3
* Faust version 1.6.1
* Operating system Windows
* Kafka version  2.12-2.2.0

**Comment by ask** [comment]
Thanks!

I actually found the bug, it is in the expiry of old items in the acked index.
We find the next commit offset by looking at consecutive numbers, and in this
case when we remove from the index it becomes consecutive.

I don't think this always happens, but definitely it does in this case :(

I fixed the issue in master

**Comment by ask** [comment]
This is fixed in 1.7.0
